### PR TITLE
Mo 120 create custom departments module simple department

### PR DESCRIPTION
**Pull request by AndriiK139MCAF**
Created a simple department notebook from different module just to make proof of concept

### CODE REVIEW

File: crm_custom_departments/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+    'name': "CRM departments",
+    'summary': "Service for handling client and household management",
```
Comment (MORPSmcaf) [inline crm_custom_departments/__manifest__.py:3]: rename to Departments management

File: crm_custom_departments/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    'author': "MCAF",
+    'website': "https://mcaf.nb.ca/en/",
+    'author': "MCAF",
+    'category': "CRM Internal",
+    'license': 'OPL-1',
+    "application": True,
+    "installable": True,
+    'version': '16.0.1.0.0',
+    'depends': ['base', 'mail', 'custom_crm_client'],
```
Comment (MORPSmcaf) [inline crm_custom_departments/__manifest__.py:12]: remove mail if you dont need it

File: crm_custom_departments/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+    'data': [
+        'security/ir.model.access.csv',
+        'views/views.xml',
+        'views/menu.xml',
+    ],
+    'demo': [
+        'demo/demo.xml',
```
Comment (MORPSmcaf) [inline crm_custom_departments/__manifest__.py:20]: remove 'demo/demo.xml', if you dont use it

File: crm_custom_departments/controllers/__init__.py
Comment (MORPSmcaf) [inline crm_custom_departments/controllers/__init__.py:1]: remove, if you dont use it

File: crm_custom_departments/controllers/controllers.py
Comment (MORPSmcaf) [inline crm_custom_departments/controllers/controllers.py:1]: remove file, if you dont use it

File: crm_custom_departments/demo/demo.xml
Comment (MORPSmcaf) [inline crm_custom_departments/demo/demo.xml:1]: remove file, if you dont use it

File: crm_custom_departments/models/department.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
+from odoo import models, fields, _
+
+
+class Department_test(models.Model):
```
Comment (MORPSmcaf) [inline crm_custom_departments/models/department.py:4]: rename to Department

File: crm_custom_departments/models/department.py [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+    _name = 'department_test'
```
Comment (MORPSmcaf) [inline crm_custom_departments/models/department.py:5]: rename to crm.department

File: crm_custom_departments/models/department.py [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+    _description = "Model to create departments"
```
Comment (MORPSmcaf) [inline crm_custom_departments/models/department.py:6]: rename to CRM department

File: crm_custom_departments/models/department.py [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+
+    name = fields.Char(string='Department Name', required=True)
+    test_field = fields.Char(string='test', required=True)
+
+class Addition_client(models.Model):
```
Comment (MORPSmcaf) [inline crm_custom_departments/models/department.py:11]: rename to  AdditionClient

File: crm_custom_departments/models/department.py [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+    _inherit = 'crm.client'
+    department_ids = fields.Many2many('department_test', string='Departments')
```
Comment (MORPSmcaf) [inline crm_custom_departments/models/department.py:13]: rename department_test to crm.department

File: crm_custom_departments/security/ir.model.access.csv [hunk 9]
```diff
@@ -0,0 +1,4 @@
+"id","name","model_id:id","group_id:id","perm_read","perm_write","perm_create","perm_unlink"
+access_department_test,access_department_test,model_department_test,base.group_user,1,1,1,1
```
Comment (MORPSmcaf) [inline crm_custom_departments/security/ir.model.access.csv:2]: rename department_test

File: crm_custom_departments/views/views.xml [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+                                </tree>
+                            </field>
+                        </page>
+                    </notebook>
+                </form>
+            </xpath>
+        </field>
+    </record>
+
+    <record id="view_department_test_form" model="ir.ui.view">
```
Comment (MORPSmcaf) [inline crm_custom_departments/views/views.xml:26]: rename department_test

File: crm_custom_departments/views/views.xml [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+        <field name="name">department_test.form</field>
```
Comment (MORPSmcaf) [inline crm_custom_departments/views/views.xml:27]: rename department_test

File: crm_custom_departments/views/views.xml [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+        <field name="model">department_test</field>
```
Comment (MORPSmcaf) [inline crm_custom_departments/views/views.xml:28]: rename department_test

File: crm_custom_departments/views/views.xml [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+                <sheet>
+                    <group>
+                        <field name="name"/>
+                        <field name="test_field"/>
+                    </group>
+                </sheet>
+            </form>
+        </field>
+    </record>
+    <record id="view_department_test_tree" model="ir.ui.view">
```
Comment (MORPSmcaf) [inline crm_custom_departments/views/views.xml:40]: rename department_test

File: crm_custom_departments/views/views.xml [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+        <field name="name">department_test.tree</field>
```
Comment (MORPSmcaf) [inline crm_custom_departments/views/views.xml:41]: rename department_test

File: crm_custom_departments/views/views.xml [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+                <field name="name"/>
+                <field name="test_field"/>
+            </tree>
+        </field>
+    </record>
+
+  <record id="action_department" model="ir.actions.act_window">
+    <field name="name">departments</field>
+    <field name="type">ir.actions.act_window</field>
+    <field name="res_model">department_test</field>
```
Comment (MORPSmcaf) [inline crm_custom_departments/views/views.xml:54]: rename department_test

File: crm_custom_departments/__manifest__.py (hunk 1 above)
Comment (AndriiK139MCAF) [reply crm_custom_departments/__manifest__.py:3]: Done

File: crm_custom_departments/__manifest__.py (hunk 2 above)
Comment (AndriiK139MCAF) [reply crm_custom_departments/__manifest__.py]: Done

File: crm_custom_departments/__manifest__.py (hunk 3 above)
Comment (AndriiK139MCAF) [reply crm_custom_departments/__manifest__.py]: Done

File: crm_custom_departments/controllers/__init__.py
Comment (AndriiK139MCAF) [reply crm_custom_departments/controllers/__init__.py:1]: Done

File: crm_custom_departments/views/views.xml (hunk 11 above)
Comment (AndriiK139MCAF) [reply crm_custom_departments/views/views.xml]: Rename to department

File: crm_custom_departments/views/views.xml (hunk 12 above)
Comment (AndriiK139MCAF) [reply crm_custom_departments/views/views.xml]: Rename to department

File: crm_custom_departments/views/views.xml (hunk 13 above)
Comment (AndriiK139MCAF) [reply crm_custom_departments/views/views.xml]: Rename to department

File: crm_custom_departments/views/views.xml (hunk 14 above)
Comment (AndriiK139MCAF) [reply crm_custom_departments/views/views.xml]: Rename to department

File: crm_custom_departments/views/views.xml (hunk 15 above)
Comment (AndriiK139MCAF) [reply crm_custom_departments/views/views.xml]: Rename to department

File: crm_custom_departments/models/department.py [hunk 16]
```diff
@@ -0,0 +1,14 @@
+from odoo import models, fields, _
+
+
+class Department(models.Model):
+    _name = 'department'
```
Comment (MORPSmcaf) [inline crm_custom_departments/models/department.py:5]: rename to crm.department

File: crm_custom_departments/__init__.py
Comment (MORPSmcaf) [inline crm_custom_departments/__init__.py:1]: rename crm_custom_departments to custom_crm_departments

Comment (AndriiK139MCAF) [reply crm_custom_departments/__init__.py:1]: done

File: crm_custom_departments/models/department.py (hunk 16 above)
Comment (AndriiK139MCAF) [reply crm_custom_departments/models/department.py]: done

**Review by MORPSmcaf — APPROVED** [review]

**Pull request merged**